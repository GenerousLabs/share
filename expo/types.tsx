export type BottomTabParamList = {
  TabOne: undefined;
  TabTwo: undefined;
};

export type TabOneParamList = {
  TabOneScreen: undefined;
};

export type TabTwoParamList = {
  TabTwoScreen: undefined;
};
