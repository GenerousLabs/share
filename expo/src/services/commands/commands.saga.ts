import { all } from "typed-redux-saga/macro";
import createReadAuthTokenforRepoSaga from "./sagas/createReadAuthTokenForRepo.saga";

export {
  sagaAction as createReadAuthTokenForRepoSagaAction,
  effectSaga as createReadAuthTokenForRepoEffect,
} from "./sagas/createReadAuthTokenForRepo.saga";

export default function* commandsSaga() {
  yield all([createReadAuthTokenforRepoSaga()]);
}
