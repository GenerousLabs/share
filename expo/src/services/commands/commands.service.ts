import * as zod from "zod";
import {
  MINIMUM_TOKEN_LENGTH,
  READ_TOKENS_FILE_NAME,
  RepoType,
} from "../../shared.constants";
import { RepoInRedux } from "../../shared.types";
import { appendLineTofile, join } from "../fs/fs.service";
import { getRepoBasename, getRepoPath } from "../repo/repo.service";

export const _getReadAuthFilePath = (repo: RepoInRedux) => {
  const basename = getRepoBasename(repo);

  const commandsRepoPath = getRepoPath({ type: RepoType.commands });

  return join(commandsRepoPath, basename, READ_TOKENS_FILE_NAME);
};

export const _tokenSchema = zod
  .string()
  .nonempty()
  .min(MINIMUM_TOKEN_LENGTH, { message: "Token is too short #r6wall" });

export const addReadAuthTokenForRepo = async ({
  repo,
  token,
}: {
  repo: RepoInRedux;
  token: string;
}) => {
  _tokenSchema.parse(token);

  const tokenFilePath = _getReadAuthFilePath(repo);

  await appendLineTofile({
    filepath: tokenFilePath,
    line: token,
  });
};
