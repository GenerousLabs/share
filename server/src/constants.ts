import { join } from "path";

export const __DEV__ = process.env.NODE_ENV === "production";

// NOTE: We need to run `process.cwd()` here, when running it inside `join()` we
// get the `dist/` folder. No idea why. Insanity.
export const CWD = process.cwd();
export const DATA_PATH = join(CWD, "data");

export const POSTOFFICE_PATH = join(DATA_PATH, "postoffice");

export const REPOS_ROOT = join(DATA_PATH, "repos");
export const COMMANDS_REPO_NAME = "commands.git" as const;

export const REPO_TEMPLATE_PATH = join(CWD, "templates/empty-repo");

export const USER_TOKEN_PATH = join(
  REPOS_ROOT,
  "commands.git",
  "user_tokens.yaml"
);

export const ENOENT = "ENOENT" as const;

// The ID is comprised of 3 words joined with a dash, not sure what the minimum
// length of each word is though.
export const MINIMUM_POSTOFFICE_ID_LENGTH = 12;

// NOTE: These 3 are copied to `expo/src/shared.constants.ts` file and must be
// manually kept in sync in both.
export const READ_TOKENS_FILE_NAME = "read_tokens.txt" as const;
export const MINIMUM_USERNAME_LENGTH = __DEV__ ? 1 : 3;
export const MINIMUM_TOKEN_LENGTH = __DEV__ ? 1 : 20;
