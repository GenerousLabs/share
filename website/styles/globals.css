/* montserrat-regular - latin */
@font-face {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  src: url("/_static/fonts/montserrat-v15-latin-regular.eot"); /* IE9 Compat Modes */
  src: local(""),
    url("/_static/fonts/montserrat-v15-latin-regular.eot?#iefix")
      format("embedded-opentype"),
    /* IE6-IE8 */ url("/_static/fonts/montserrat-v15-latin-regular.woff2")
      format("woff2"),
    /* Super Modern Browsers */
      url("/_static/fonts/montserrat-v15-latin-regular.woff") format("woff"),
    /* Modern Browsers */ url("/_static/fonts/montserrat-v15-latin-regular.ttf")
      format("truetype"),
    /* Safari, Android, iOS */
      url("/_static/fonts/montserrat-v15-latin-regular.svg#Montserrat")
      format("svg"); /* Legacy iOS */
}

/* montserrat-700 - latin */
@font-face {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 700;
  src: url("/_static/fonts/montserrat-v15-latin-700.eot"); /* IE9 Compat Modes */
  src: local(""),
    url("/_static/fonts/montserrat-v15-latin-700.eot?#iefix")
      format("embedded-opentype"),
    /* IE6-IE8 */ url("/_static/fonts/montserrat-v15-latin-700.woff2")
      format("woff2"),
    /* Super Modern Browsers */
      url("/_static/fonts/montserrat-v15-latin-700.woff") format("woff"),
    /* Modern Browsers */ url("/_static/fonts/montserrat-v15-latin-700.ttf")
      format("truetype"),
    /* Safari, Android, iOS */
      url("/_static/fonts/montserrat-v15-latin-700.svg#Montserrat")
      format("svg"); /* Legacy iOS */
}

html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, Montserrat, BlinkMacSystemFont, Segoe UI, Roboto,
    Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: underline;
  font-weight: bold;
}

p {
  font-size: 16px;
}

* {
  box-sizing: border-box;
}

.appWrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
}

@media screen and (max-width: 580px) {
  .appContainer {
    max-width: 580px;
    flex-grow: 1;
    padding: 15px;
  }
}
@media screen and (min-width: 581px) {
  .appContainer {
    width: 580px;
    flex-grow: 1;
  }
}

header {
  position: sticky;
  width: 100vw;
  top: 0;
  background-color: #212121;
  color: white;
  display: flex;
  justify-content: center;
  margin-bottom: 20px;
}
header h1 {
  text-transform: uppercase;
  font-size: 26px;
}

.logoContainer {
  display: flex;
  justify-content: flex-start;
}

h2 {
  font-size: 36px;
}

.h2SecondLine {
  font-size: 60px;
}

h3 {
  font-size: 24px;
  font-weight: normal;
  text-transform: uppercase;
  margin-top: 45px;
}

h4 {
  font-size: 24px;
  font-weight: normal;
  text-transform: uppercase;
}

hr {
  height: 1px;
  border: 0;
  border-top: 1px solid #c4c4c4;
  margin-top: 34px;
  margin-bottom: 34px;
}

.buttonLinkP {
  font-size: 14px;
  padding: 15px;
  text-align: center;
  border: 1px solid #212121;
  width: 100%;
  background-color: white;
}

.buttonLinkP a {
  text-transform: uppercase;
  text-decoration: none;
}

.alternateLinkP {
  font-size: 12px;
  text-align: center;
}

.alternateLinkP a {
  font-weight: normal;
}

.blackHighlight {
  background-color: #212121;
  color: white;
  padding: 20px;
  margin-top: 30px;
  margin-bottom: 30px;
}

.nobreak {
  white-space: nowrap;
}
