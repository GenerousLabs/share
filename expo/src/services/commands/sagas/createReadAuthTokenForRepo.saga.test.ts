import { describe, expect, it } from "@jest/globals";
import { testSaga } from "redux-saga-test-plan";
import { call, putResolve } from "redux-saga/effects";
import { RepoType } from "../../../shared.constants";
import { RepoInRedux } from "../../../shared.types";
import { commitAllSagaAction } from "../../repo/sagas/commitAll.saga";
import { addReadAuthTokenForRepo } from "../commands.service";
import {
  createReadAuthTokenForRepoSagaAction,
  effect,
} from "./createReadAuthTokenForRepo.saga";

describe("createReadAuthTokenForRepo.saga", () => {
  it("Creates a new invitation and returns a token #QkEGyv", () => {
    const repo = {
      id: "repo1",
      uuid: "repo1",
      path: "/repo1/",
      headCommitObjectId: "",
      lastFetchTimestamp: 0,
      bodyMarkdown: "",
      isReadOnly: false,
      name: "repo1",
      remoteUrl: "",
      title: "repo1",
      type: RepoType.connection,
    } as RepoInRedux;
    const commandRepo = {
      ...repo,
      id: "commands",
      uuid: "commands",
      path: "commands",
      name: "commands",
      title: "commands",
      type: RepoType.commands,
    };

    // NOTE: We need to define this as '' here otherwise TypeScript complains
    // below that it is used before being defined.
    let token: string = "";

    testSaga(
      effect,
      createReadAuthTokenForRepoSagaAction({
        repoId: "repo1",
        connectionId: "connection1",
      })
    )
      .next()
      .inspect((val) => expect(val).toMatchSnapshot())
      .next(repo)
      .inspect((val) => expect(val).toMatchSnapshot())
      .next(commandRepo)
      .inspect((callEffect) => {
        const matchEffect = call(addReadAuthTokenForRepo, {
          repo,
          token: (expect.any(String) as unknown) as string,
        });
        expect(callEffect).toEqual(matchEffect);
        token = (callEffect as any).payload.args[0].token;
      })
      .next()
      .inspect((callEffect) => {
        const matchEffect = putResolve(
          commitAllSagaAction({
            repoId: commandRepo.id,
            message: (expect.any(String) as unknown) as string,
          })
        );
        expect(callEffect).toEqual(matchEffect);
      })
      .next()
      .returns({ token });
  });

  // it('Returns the token when passed to store.dispath #8iqXoi', () => {
  //   const store = createMockStore()
  // })
});
