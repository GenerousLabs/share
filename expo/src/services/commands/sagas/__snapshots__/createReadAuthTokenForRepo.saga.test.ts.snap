// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createReadAuthTokenForRepo.saga Creates a new invitation and returns a token #QkEGyv 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "repo1",
    ],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`createReadAuthTokenForRepo.saga Creates a new invitation and returns a token #QkEGyv 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;
